MerlinAU.sh

```diff
@@ -4487,6 +4488,197 @@ FW_InstalledVers="$(_GetCurrentFWInstalledShortVersion_)"
+}
+
+##-------------------------------------##
+## Added by Martinski W. [2024-May-03] ##
+##-------------------------------------##
+_GetFileSelection_()
+{
+   if [ $# -eq 0 ] || [ -z "$1" ] ; then return 1 ; fi
+
+   if [ $# -lt 2 ] || [ "$2" != "-MULTIOK" ]
+   then indexType="" ; else indexType="$2" ; fi
+
+   theFilePath=""  theFileName=""  fileTemp=""
+   fileCount=0  fileIndex=0  multiIndex=false
+   local sourceDirPath="$FW_LOG_DIR"
+   local maxFileCount=20
+
+   printf "\n${1}\n[Directory: ${GRNct}${sourceDirPath}${NOct}]\n\n"
+
+   while IFS="$(printf '\n\t')" read -r backupFilePath
+   do
+       fileCount=$((fileCount+1))
+       fileVar="file_${fileCount}_Name"
+       eval file_${fileCount}_Name="${backupFilePath##*/}"
+       printf "${GRNct}%3d${NOct}. " "$fileCount"
+       eval echo "\$${fileVar}"
+       [ "$fileCount" -ge "$maxFileCount" ] && break
+   done <<EOT
+$(_list2_ -1t "$theLogFilesMatch" 2>/dev/null)
+EOT
+
+   echo
+   _GetFileSelectionIndex_ "$fileCount" "$indexType"
+
+   if [ "$fileIndex" = "ALL" ] || [ "$fileIndex" = "NONE" ]
+   then theFilePath="$fileIndex" ; return 0 ; fi
+
+   if [ "$indexType" = "-MULTIOK" ] && "$multiIndex"
+   then
+       for index in $fileIndex
+       do
+           fileVar="file_${index}_Name"
+           eval fileTemp="\$${fileVar}"
+           if [ -z "$theFilePath" ]
+           then theFilePath="${sourceDirPath}/$fileTemp"
+           else theFilePath="${theFilePath}|${sourceDirPath}/$fileTemp"
+           fi
+       done
+   else
+       fileVar="file_${fileIndex}_Name"
+       eval theFileName="\$${fileVar}"
+       theFilePath="${sourceDirPath}/$theFileName"
+   fi
+   return 0
+}
+
+##----------------------------------------##
+## Modified by Martinski W. [2024-May-04] ##
+##----------------------------------------##
+_CheckForUpdateLogFiles_()
+{
+   theLogFilesMatch="${FW_LOG_DIR}/${MODEL_ID}_FW_Update_*.log"
+   theFileCount="$(_list2_ -1 "$theLogFilesMatch" 2>/dev/null | wc -l)"
+   
+   if [ ! -d "$FW_LOG_DIR" ] || [ "$theFileCount" -eq 0 ]
+   then
+       updateLogFileFound=false
+       return 1
+   fi
+   chmod 444 "${FW_LOG_DIR}"/${MODEL_ID}_FW_Update_*.log
+   updateLogFileFound=true
+   return 0
+}
+
+##----------------------------------------##
+## Modified by Martinski W. [2024-May-04] ##
+##----------------------------------------##
+_ViewUpdateLogFile_()
+{
+   local theFilePath=""  theFileCount  theLogFilesMatch  retCode
+
+   if ! _CheckForUpdateLogFiles_
+   then
+       printf "\n${REDct}**ERROR**${NOct}: Log file(s) [$theLogFilesMatch] NOT FOUND."
+       return 1
+   fi
+   printf "\n---------------------------------------------------"
+   _GetFileSelection_ "Select a log file to open:"
+
+   if [ "$theFilePath" = "NONE" ] || [ ! -f "$theFilePath" ]
+   then return 1 ; fi
+
+   printf "\nLog file to view:\n${GRNct}${theFilePath}${NOct}\n"
+   printf "\n[Press ${REDct}<Esc>:q!${NOct} to quit viewer]\n"
+   _WaitForEnterKey_
+
+   vi "$theFilePath"
```

Comment:
Instead of vi, I think maybe we should use less.
It's closer in behavior to the "more" we use for the changelog file.

vi is a full editor, I don't think we need an editor myself, I think a simple user as you identified just need to view the changelog file and have an easy way to close.

Now I am familiar with <Esc>:q! myself, but my concern is right now is a newer user as you mentioned may get lost if they don't read that or remember it, it also doesn't line up with the changelog option which is simply "Q" to close.

If we use "less" at least they would have similar closure behavior to the CL option and it's only a viewer not an editor so no risk of them using features they don't know. It's just a simple viewer that allows for navigate both forwards and backwards through the file and that's all I think we need for the MerlinAU logs.